include/rmm/device_buffer.hpp

```diff
@@ -201,8 +201,7 @@ class device_buffer {
       // If the current capacity is large enough and the resources are
       // compatible, just reuse the existing memory
       if ((capacity() > other.size()) and _mr->is_equal(*other._mr)) {
-        set_stream(other.stream());
-        resize(other.size());
+        resize(other.size(), other.stream());
```

Comment:
`resize()` calls `set_stream()` internally. This was resetting `_stream` to the default 0 instead of `other.stream()`.

Reply:
Hm, should this be on `other.stream()` or `this->stream()`?

Reply:
The documentation says
> new memory will be allocated using `other.memory_resource()` on `other.stream()`.
However, maybe it shouldn't be using either since it hides the fact that this is a stream ordered operation. Could be replaced with a `clone()` method that takes a parameter.

Reply:
Yeah, I long debated removing the copy assignment operator all together. Though that would mean we'd have to remove/modify `cudf::column`s copy assignment operator.

Reply:
Well, if `column` and `column_view` each hold on to a dedicated stream that their operations are to be ordered on, it wouldn't be a problem :wink:

Reply:
Back to the earlier question though, I think `other.stream()` makes more sense in this case. This is the `capacity() > other.size()` branch, so `resize()` is only going to update the `_size` member. The following `copy()` is overwriting the old data with `other.data()`, and it seems appropriate to do so in `other.stream()`, assuming any pending writes to `other.data()` that we want to copy after would be in that stream.

Reply:
> Well, if column and column_view each hold on to a dedicated stream that their operations are to be ordered on, it wouldn't be a problem

That just moves the same problem up a layer. Instead of `device_buffer` holding on to a stream, now `column` does. 

> seems appropriate to do so in other.stream(), assuming any pending writes to other.data() that we want to copy after would be in that stream.

This just reminds me why I really, really dislike `device_buffer` having a copy assignment operator. It's a stream ordered operation that doesn't take a stream argument. Unless the user is really paying attention, they're not going to know that this is a stream ordered copy.